enhancements/hypershift/hostedClusterInfrastructure.md

```diff
@@ -0,0 +1,166 @@
+HostedClusterInfrastructure would be used to provision or track Cloud Provider infrastructure objects that can be used when provisioning HostedClusters using those providers.
+
+## Glossary
+- Untangling HypershiftDeployment, HostedCluster and NodePool Google document - [Google Document - Option D](https://docs.google.com/document/d/1VabhUQa_uZlWO6mf9UHS5lGtnH_X915cih4ARJbNibc/edit#heading=h.iuee7tji8zz4)
+
+## Motivation
+ACM and MCE primary Cloud Provider use cases already provision these resources and manage these resources, this provides feature parity with Standalone flows that ACM Cluster Administrator's are used to.  This also allows to additional opporunities:
+  1. Infrastructure can be pre-created using an ACM Cloud Provider credential or by pre-populating the Custom Resource with values and applying it to the ACM/MCE hub. These can then be consumed by the ACM/MCE UI or CLI, and is a nice option, as the AWS credential does not need to be exposed to the user creating the HostedCluster.
+  2. You can re-use the HostedClusterInfrastructure resource between HostedClusters. This is especially useful in Azure, where it takes longer to create infrastructure then the hosted control plane.
+
+### User Stories
+- As a Cluster Administrator I want to create one or more HostedClusterInfrastructure resources for my team to consume.
+  
+  This approach pre-creates the HostedClusterInfrastructure before the consumer creates the HostedCluster and NodePool resources. This approach has two advantage for ACM use cases:
+  * The first advantage is that and Administrator can create the infrastructure resources using either their account or a service account stored in the ACM Cloud Provider credential (AWS or Azure cred).  
+  The result HostedClusterInfrastructure object can be given get/list verbs to the consumer who wants to create a cluster. 
+  This consumer would then only need CRUD for HostedCluster and NodePool and be able to use the console to pick a HostedClusterInfrastructure resource and then create a HostedCluster.  No AWS credential needs to be passed to the consumer.  
+  * The second is that for slow cloud providers like Azure, the infrastructure can be pre-created to improve cluster deployment time (in dev, test, and pipeline scenarios). 
+  It also seems possible to re-use the infrastructure between clusters, so treating HostedClusterInfrastructure as an entity outside the hostedCluster promotes scenarios of re-use (saves time, especially in Azure. Most likely a dev or pipeline user case).
+  * The third is that the HostedClusterInfrastructure YAML can be committed to Git and applied using ArgoCD, Flux or ACM Subscriptions to create a gitOps flow for infrastructure creation and deletion.
+  * The fourth is to apply HostedClusterInfrastructure with value already present and either include or not include the cloud provider credential.  
```

Comment:
>* The fourth is to apply HostedClusterInfrastructure with value already present and either include or not include the cloud provider credential.

to clarify Is this to "adopt" existing infra?

Reply:
done.